src/main/java/ogc/rs/database/DatabaseServiceImpl.java

```diff
@@ -475,36 +476,52 @@ public Future<JsonObject> getStacItemById(String collectionId, String stacItemId
   }
 
   @Override
-  public Future<JsonObject> stacItemSearch(Map<String, String> queryParams) {
+  public Future<JsonObject> stacItemSearch(StacItemSearchParams params) {
     LOGGER.debug("stacItemSearch");
+    
     Promise<JsonObject> result = Promise.promise();
-    Collector<Row, ? , List<JsonObject>> collector = Collectors.mapping(Row::toJson, Collectors.toList());
-    String[] collectionIds = queryParams.get("collections").split(",");
-    FeatureQueryBuilder featureQuery = new FeatureQueryBuilder(collectionIds);
-    featureQuery.setLimit(Integer.parseInt(queryParams.getOrDefault("limit","10")));
-    featureQuery.setOffset(Integer.parseInt(queryParams.getOrDefault("offset", "0")));
+    final String STAC_ITEMS_DATETIME_KEY = "properties ->> 'datetime'";
+
+    Collector<Row, ?, List<JsonObject>> collector =
+        Collectors.mapping(Row::toJson, Collectors.toList());
+
+    FeatureQueryBuilder featureQuery = new FeatureQueryBuilder();
 
-    if (queryParams.containsKey("bbox")){
+    featureQuery.setLimit(params.getLimit());
+    featureQuery.setOffset(params.getOffset());
+
+    if (!params.getBbox().isEmpty()) {
       featureQuery.setBboxCrsSrid(String.valueOf(DEFAULT_CRS_SRID));
-      featureQuery.setBbox(queryParams.get("bbox"), String.valueOf(DEFAULT_CRS_SRID));
+      featureQuery.setBbox(
+          params.getBbox().stream().map(i -> i.toString()).collect(Collectors.joining(",")),
+          String.valueOf(DEFAULT_CRS_SRID));
+    }
+
+    if (params.getDatetime() != null) {
+      featureQuery.setDatetimeKey(STAC_ITEMS_DATETIME_KEY);
+      featureQuery.setDatetime(params.getDatetime());
     }
-    //TODO: convert individual DB calls to a transaction
-    if (queryParams.containsKey("datetime")) {
-      featureQuery.setDatetime(queryParams.get("datetime"));
+
+    if (!params.getCollections().isEmpty()) {
+      featureQuery.setStacCollectionIds(params.getCollections().toArray(String[]::new));
     }
 
-    if (queryParams.containsKey("ids")) {
-      featureQuery.setItemIds(queryParams.get("ids"));
+    if (!params.getIds().isEmpty()) {
+      featureQuery.setStacItemIds(params.getIds().toArray(String[]::new));
     }
 
-    if (queryParams.containsKey("intersects")) {
-      featureQuery.setGeometryIntersects(queryParams.get("geometry"));
+    if (params.getIntersects() != null) {
+      featureQuery.setStacIntersectsGeom(params.getIntersects());
     }
+    
+    Tuple tuple = Tuple.tuple();
+    String builtQuery = featureQuery.buildItemSearchSqlString(tuple);
 
     JsonObject resultJson = new JsonObject();
+    
     client.withConnection(conn ->
-        conn.preparedQuery(featureQuery.buildItemSearchSqlString())
-          .collecting(collector).execute().map(SqlResult::value)
+        conn.preparedQuery(builtQuery)
+          .collecting(collector).execute(tuple).map(SqlResult::value)
```

Comment:
Where is this tuple getting populated from? I see it's been populated in the `FeatureQueryBuilder` class but it's not returned anywhere. Am I missing something?

Reply:
- The `tuple` object is passed into `FeatureQueryBuilder` and is getting modified there. Since the same object is being referenced in `DatabaseServiceImpl`, it works. This would not have worked if the `Tuple` class was immutable.
- It's not great, I didn't know how else to get a tuple back from `FeatureQueryBuilder` without writing a lot of code then. The only 'nice' way that I can think of doing this is to have a custom holder class that has 2 fields `String sql` and `Tuple tuple` that gets returned by the `buildItemSearchSqlString` method. I can make this change if you'd like. 
- If we had Apache Commons Lang as an import, we could have used [Pair](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/tuple/Pair.html)

Reply:
If the Tuple object is getting populated and destroyed properly, I think it's okay to have it for now. If we run into any sort of issues later on, we can think of these strategies.